src/core/trtserver.h

```diff
@@ -108,6 +109,64 @@ TRTSERVER_EXPORT const char* TRTSERVER_ErrorCodeString(TRTSERVER_Error* error);
 /// \return The error message.
 TRTSERVER_EXPORT const char* TRTSERVER_ErrorMessage(TRTSERVER_Error* error);
 
+/// TRTSERVER_ResponseAllocator
+///
+/// Object representing a memory allocator for inference response
+/// tensors.
+///
+
+/// Allocation regions.
+typedef enum trtserver_allocatorregion_enum {
+  TRTSERVER_MEMORY_CPU,
+  TRTSERVER_MEMORY_GPU
+} TRTSERVER_Allocator_Region;
+
+/// Type for allocation function that allocates a buffer to hold a
+/// result tensor. Return in 'buffer' a pointer to the contiguous
+/// memory block of size 'byte_size' for result tensor called
+/// 'tensor_name'. The buffer must be allocated in the memory region
+/// identified by 'region' and 'region_id'. The 'userp' data is the
+/// same as what is supplied in the call to
+/// TRTSERVER_ServerInferAsync.
+///
+/// The function may return 'buffer' == nullptr to indicate that an
+/// allocation in the requested 'region' is not possible. In this case
+/// the function may be called again for the same 'tensor_name' but
+/// with a different 'region'.
```

Comment:
Is the returning Error object a nullptr in this case?

Reply:
Yes, any non-null error returned will indicate that something went wrong and the server will fail the request.